src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/OptimizedInboxTextEncoder.AdvSimd64.cs

```diff
@@ -0,0 +1,257 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+
+namespace System.Text.Encodings.Web
+{
+    internal sealed partial class OptimizedInboxTextEncoder
+    {
+        private unsafe nuint GetIndexOfFirstByteToEncodeAdvSimd64(byte* pData, nuint lengthInBytes)
+        {
+            Debug.Assert(AdvSimd.Arm64.IsSupported);
+            Debug.Assert(BitConverter.IsLittleEndian);
```

Comment:
Can we make that assumption? To my knowledge arm64 is bi-endian.

Reply:
Good catch. I'll guard this at the call site. If we get a big-endian machine in our test lab then we can revisit this code.

Reply:
I just found out we apparently have one in the runtimelab CI.